### PR TITLE
reference: update DM upgrade doc, add v1.0.2 / v1.0.1 

### PR DESCRIPTION
**Pull request by ericsyh**
### What is changed, added or deleted? 

Remove the previous DM version in the upgrade doc and add v1.0.2, v1.0.1 upgrade steps.

### CODE REVIEW

File: dev/reference/tools/data-migration/dm-upgrade.md [hunk 1]
```diff
@@ -19,180 +19,117 @@ category: reference
...
-
-从此版本开始，上述两项配置信息被移除，并增加了如下配置信息：
-
-- `source_id`：存在于 `inventory.ini` 中，用于标识一个上游 MySQL 实例或一个主从复制组。
-- `source-id`：存在于任务配置文件的 `mysql-instances` 中，其取值与 `inventory.ini` 中的 `source_id` 对应。
-
-> **注意：**
->
-> 如果需要确保已有任务存储在下游数据库的断点信息能继续被使用，`source_id`/`source-id` 的值需要与对应 DM-worker 变更前的 `instance-id` 一致。
+从此版本开始，将对所有的配置进行严格检查，遇到不识别的配置会报错，以确保用户始终准确地了解自己的配置。
```
Comment (yikeke) [inline dev/reference/tools/data-migration/dm-upgrade.md]:
```suggestion
从此版本开始，将对所有的配置进行严格检查，遇到无法识别的配置会报错，以确保用户始终准确地了解自己的配置。
```

File: dev/reference/tools/data-migration/dm-upgrade.md [hunk 2]
```diff
@@ -19,180 +19,117 @@ category: reference
...
-    instance-id: "instance118-4306" # 此值具有唯一性，当保存 checkpoint、配置和其他信息时，作为 ID 使用。
-    ```
-
-    变更为
-
-    ```yaml
-    source-id: "instance118-4306" # 如需要重用之前任务的 checkpoint，需要与原 `instance-id` 取值一致。
-    ```
+启动 DM-master 或 DM-worker 前，必须确保已经删除废弃的配置信息，且没有多余的配置项，否则会启动失败。可根据失败信息删除多余的配置。
+可能遗留的废弃配置:
```
Comment (yikeke) [inline dev/reference/tools/data-migration/dm-upgrade.md]:
```suggestion
可能遗留的废弃配置：
```

File: dev/reference/tools/data-migration/dm-upgrade.md [hunk 3]
```diff
@@ -19,180 +19,117 @@ category: reference
...
+- 直接管理到下游的 DB 连接而不是使用内置连接池，优化 SQL 错误处理与重试
+- 修复 DM-worker 进程启动时、执行 DML 失败时可能 panic 的 bug
+- 修复执行 sharding DDL（如 `ADD INDEX`）超时后可能造成后续 sharding DDL 无法正确协调的 bug
+- 修复了有部分 DM-worker 不可访问时无法 `start-task` 的 bug
+- 完善了对 1105 错误的自动重试策略
 
 ### 升级操作示例
 
-1. 下载新版本 DM-Ansible, 确认 `inventory.ini` 文件中 `dm_version = v1.0.0`
+1. 下载新版本 DM-Ansible, 确认 `inventory.ini` 文件中 `dm_version = v1.0.2`
```
Comment (csuzhangxc) [inline dev/reference/tools/data-migration/dm-upgrade.md]: 这文档前面有一句 `本文档主要介绍各不完全兼容的 DM (Data Migration) 版本间的升级操作步骤。`。但 v1.0.1 和 v1.0.2 可以算是兼容的，这里要改一下描述吗？

Comment (ericsyh) [reply dev/reference/tools/data-migration/dm-upgrade.md]: 要改一下的。。

Comment (ericsyh) [reply dev/reference/tools/data-migration/dm-upgrade.md]: Update in https://github.com/pingcap/docs-cn/pull/1998/commits/b795d9175ce31756d283ebb81d0dc2af1f5a00ec

**Pull request merged**